client/starwhale/core/model/model.py

```diff
@@ -694,10 +689,117 @@ def serve(
         port: int,
     ) -> None:
         _model_config = cls.load_model_config(workdir / model_yaml, workdir)
-        svc = cls._get_service(_model_config.run.handler, workdir)
+        svc = ServeHandlerParser.get_service(_model_config.run.handler, workdir)
         svc.serve(host, port, _model_config.name)
 
 
+class HandlerParser:
+    sw_dir: str = SW_AUTO_DIRNAME
+
+    def __init__(self, workdir: Path, handler: t.Any, target_dir: Path) -> None:
+        self.workdir = workdir
+        self.handler = handler
+        self.target_dir = target_dir
```

Comment:
Should we use the target_dir as the finna yaml parent dir?

Reply:
fixed